tests/lint/sconscript

```diff
@@ -0,0 +1,45 @@
+###
+# This file is a part of the NVDA project.
+# URL: http://www.nvaccess.org/
+# Copyright 2019 NV Access Limited.
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 2.0, as published by
+# the Free Software Foundation.
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+# This license can be found at:
+# http://www.gnu.org/licenses/old-licenses/gpl-2.0.html
+###
+
+import os
+import sys
+
+Import("env")
+
+# Add system path to get access to git.
+externalEnv = Environment(ENV = {'PATH' : os.environ['PATH']})
+
+# Create a unified diff. Written to file for ease of manual inspection
+diffTarget = externalEnv.Command(
+	"current.diff",
+	None,
+	[["git", "diff", "-u", "HEAD", ">", '$TARGET']]

```

Comment:
I wonder whether it would be possible to make a test against the base branch. A quick Google search turns out that this is not very easy. May be hardcode this to master of upstream (not to local master). See https://stackoverflow.com/questions/29810331/is-there-a-quick-way-to-git-diff-from-the-point-or-branch-origin

```suggestion
	[["git", "diff", "-u", "origin/master...HEAD", ">", '$TARGET']]
```

Reply:
Yeah I might have a play with some of these options. This was really only provided as a convenience/example for developers. Note that appveyor knows the base branch and can do the diff more directly. 

Perhaps the clearest way is if we require the "upstream" to be set and generate the diff against that: `git diff @{u}...HEAD`. We can give an error message if the upstream is not set.

Reply:
> Perhaps the clearest way is if we require the "upstream" to be set and generate the diff against that: `git diff @{u}...HEAD`. We can give an error message if the upstream is not set.

Mm, but if I create a topic branch and push that, setting my private fork as the upstream for that particular branch which I usually do, then `git diff @{u}...HEAD` will likely compare against upstream of the currently checked out branch. This is also not what we want I assume?

Reply:
Yeah, you are right, every time you push this will be resetting the diff, not what we want. Instead, I think I will just require a 'ref' (branch / commit / tag) as an argument to scons, and we will do it against that branch.

Reply:
Awesome idea